Composite/Composite.Cs.Tests/Composites/ForkTests.cs

```diff
@@ -21,9 +17,91 @@ public void TransformationTest()
                 C.Value( new Simple { Number = 6, } ),
             });
 
-            var result = obj.Fork(x => x.Number == 1, _ => new[]{new Simple { Number = 4, }, new Simple { Number = 5, }});
+            var result = obj.Fork(x => x.Number == 1, _ => new[] { new Simple { Number = 4, }, new Simple { Number = 5, } });
 
             Assert.Equal("[ [ 4, 5 ], 6 ]", result.AsString(x => x.Number.ToString()));
         }
+
+        [Fact]
+        public void LazyInputTest()
+        {
+            var obj = C.Composite(new[] {
+                C.Composite(new[]
+                {
+                    C.Value(new Simple { Number = 1, }),
+                    C.Value(new Simple { Number = 2, }),
+                    C.Value(new Simple { Number = 3, })
+                }.AsLimited(2)),
```

Comment:
Preferrably, we only use in tests the functions that have already been tested themselves. `AsLimited()` enumerable extension and 'AsString()' `Composite<T>` extension are advised for moving to the library, covering with tests.